From f498019527eda13e52e23ab5d9d56146a37af796 Mon Sep 17 00:00:00 2001
From: "Jan Alexander Steffens (heftig)" <jan.steffens@gmail.com>
Date: Tue, 13 Oct 2015 21:00:57 +0200
Subject: [PATCH] filemonitor: Take a ref on the GDBusConnection

The initial ref is owned by the thread-local connection cache.
We unref the connection when we finalize - oops.
---
 client/gdaemonfilemonitor.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/client/gdaemonfilemonitor.c b/client/gdaemonfilemonitor.c
index 313b06f..644aecb 100644
--- a/client/gdaemonfilemonitor.c
+++ b/client/gdaemonfilemonitor.c
@@ -181,8 +181,8 @@ g_daemon_file_monitor_new (const char *remote_id,
   daemon_monitor->remote_id = g_strdup (remote_id);
   daemon_monitor->remote_obj_path = g_strdup (remote_obj_path);
 
-  daemon_monitor->connection = _g_dbus_connection_get_sync (daemon_monitor->remote_id, NULL, &error);
-  if (daemon_monitor->connection == NULL)
+  connection = _g_dbus_connection_get_sync (daemon_monitor->remote_id, NULL, &error);
+  if (connection == NULL)
     {
       g_printerr ("Error getting connection for monitoring: %s (%s, %d)\n",
                   error->message, g_quark_to_string (error->domain), error->code);
@@ -190,6 +190,8 @@ g_daemon_file_monitor_new (const char *remote_id,
     }
   else
     {
+      daemon_monitor->connection = g_object_ref (connection);
+
       if (!g_dbus_interface_skeleton_export (G_DBUS_INTERFACE_SKELETON (daemon_monitor->skeleton),
                                              daemon_monitor->connection,
                                              daemon_monitor->object_path,
-- 
2.6.1

